app/oauth.py

```diff
@@ -10,73 +13,86 @@
+        data=json.dumps(
+            {
+                "client_id": github_client_id,
+                "client_secret": github_client_secret,
+                "code": github_code,
+            }
+        ),
+        headers={"content-type": "application/json"},
+    )
+    assert (
+        github_response.status_code == 200
+    ), f"ERROR: {github_response.status_code}, {github_response.text}"
+    token = re.search(
+        r"access_token=([a-zA-Z0-9]+)", github_response.content.decode("utf-8")
     )
-    assert github_response.status_code == 200, \
-        f'ERROR: {github_response.status_code}, {github_response.text}'
-    token = re.search(r"access_token=([a-zA-Z0-9]+)", github_response.content.decode('utf-8'))
     if not token:
         raise PermissionError(github_response)
     return token.group(1)
 
 
-def retrieve_github_user_info(token):
-    """
+def retrieve_github_user_info(token: str) -> str:
+    """ "
     using the access token returned by github, retrieve the user's info from the github api
-    :token: access token generated from github
+        :param token: access token generated from github
+        :return: user's info
     """
     response = requests.get(
-        'https://api.github.com/user',
-        data={'token': token},
-        headers={'Authorization': f'token {token}', 'content-type': 'application/json'}
+        "https://api.github.com/user",
+        data={"token": token},
+        headers={"Authorization": f"token {token}", "content-type": "application/json"},
     )
     return response.json()
 
 
-def convert_to_auth_token(client_id, client_secret, backend, token):
+def convert_to_auth_token(
+    client_id: str, client_secret: str, backend: str, token: str
+) -> str:
     """
-    with the previously generated access_token, use the django-rest-framework-social-oauth2
-    endpoint `/convert-token/` to authenticate the user and return a django auth token
-    :client_id: from OauthToolkit application
-    :client_secret: from OauthToolkit application
-    :backend: social authentication backend to the user ('github', 'facebook', etc)
-    :token: access token generated from the github backend
-    :return: django auth token
+    with the previously generated access_token, use the
+    django-rest-framework-social-oauth2 endpoint `/convert-token/` to
+    authenticate the user and return a django auth token
+        :param client_id: from OauthToolkit application
+        :param client_secret: from OauthToolkit application
+        :param backend: social authentication backend to the user ('github', 'facebook', etc)
+        :param token: access token generated from the github backend
+        :return: django auth token
     """
     params = {
-        'grant_type': 'convert_token',
-        'client_id': client_id,
-        'client_secret': client_secret,
-        'backend': backend,
-        'token': token,
+        "grant_type": "convert_token",
```

Comment:
Why did you have to change to double quotes?

Reply:
this was done automatically, i think by Black. I've not merged this PR yet, and I told you I had to remove autopep8 and install Black because of package version clashes.